Dockerfile

```diff
@@ -86,6 +86,20 @@ RUN wget https://bootstrap.pypa.io/get-pip.py -O - | python3
 # Install meson cross-platform build system
 RUN pip3 install meson==0.62.2
 
+# Install rustc - environment located in /root/.rustup
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | CARGO_HOME=/usr/local sh -s -- -y
+# Install rustc stable toolchain
+RUN CARGO_HOME=/usr/local rustup toolchain install stable
+RUN CARGO_HOME=/usr/local rustup default stable
+# Install all rustc targets
+RUN CARGO_HOME=/usr/local rustup target install x86_64-unknown-linux-gnu
+RUN CARGO_HOME=/usr/local rustup target install i686-unknown-linux-gnu
+RUN CARGO_HOME=/usr/local rustup target install armv5te-unknown-linux-gnueabi
+RUN CARGO_HOME=/usr/local rustup target install armv7-unknown-linux-gnueabihf
+RUN CARGO_HOME=/usr/local rustup target install armv7-unknown-linux-gnueabi
+RUN CARGO_HOME=/usr/local rustup target install aarch64-unknown-linux-gnu
+RUN CARGO_HOME=/usr/local rustup target install powerpc-unknown-linux-gnu
+
```

Comment:
Is there a difference between
rustup target add $RUST_TARGET
And
rustup target install?

Reply:
I think the install command is for the `rustup toolchain install ` and for targets you do `rustup target add`, where the toolchain is obviously for the host doing the build, and the target is the end target when cross compiling.

Reply:
Related to your first comment, and looking at `rustup target|toolchain --help`:
```
spksrc@spksrc:/tmp$ rustup toolchain --help
rustup-toolchain 
Modify or query the installed toolchains

USAGE:
    rustup toolchain <SUBCOMMAND>

FLAGS:
    -h, --help    Prints help information

SUBCOMMANDS:
    list         List installed toolchains
    install      Install or update a given toolchain
    uninstall    Uninstall a toolchain
    link         Create a custom toolchain by symlinking to a directory
    help         Prints this message or the help of the given subcommand(s)
```
vs `target`:
```
$ rustup target --help
rustup-target 
Modify a toolchain's supported targets

USAGE:
    rustup target <SUBCOMMAND>

FLAGS:
    -h, --help    Prints help information

SUBCOMMANDS:
    list      List installed and available targets
    add       Add a target to a Rust toolchain
    remove    Remove a target from a Rust toolchain
    help      Prints this message or the help of the given subcommand(s)
```
So while `install` does work, the actual call should be `add`.  Will change that.

Reply:
Yeah I wasn't looking at the actual rust clip help, I was reading their docs. They worded it a little difference on the rust book, so the toolchain I assumed was to build host tools, and the targets were for cross.


The env file just sets some variables for the includes and linker and path afaik.